anymail/backends/sendinblue.py

```diff
@@ -0,0 +1,200 @@
+        parsed_response = self.deserialize_json_response(response, payload, message)
+        try:
+            message_id = parsed_response['messageId']
+        except (KeyError, TypeError):
+            raise AnymailRequestsAPIError("Invalid SendinBlue API response format",
+                                          email_message=message, payload=payload, response=response,
+                                          backend=self)
+
+        status = AnymailRecipientStatus(message_id=message_id, status="queued")
+        return {recipient.addr_spec: status for recipient in payload.all_recipients}
+
+
+class SendinBluePayload(RequestsPayload):
+
+    def __init__(self, message, defaults, backend, *args, **kwargs):
+        self.all_recipients = []  # used for backend.parse_recipient_status
+        self.template_id = None
+
+        http_headers = kwargs.pop('headers', {})
+        http_headers['api-key'] = backend.api_key
+        http_headers['Content-Type'] = 'application/json'
+
+        super(SendinBluePayload, self).__init__(message, defaults, backend, headers=http_headers, *args, **kwargs)
+
+    def get_api_endpoint(self):
+        if self.template_id:
+            return "smtp/templates/%s/send" % (self.template_id)
+        else:
+            return "smtp/email"
+
+    def init_payload(self):
+        self.data = {  # becomes json
+            'sender': list(),
+            'headers': dict()
+        }
+
+    def serialize_data(self):
+        """Performs any necessary serialization on self.data, and returns the result."""
+
+        headers = self.data["headers"]
+        if "Reply-To" in headers:
+            # Reply-To must be in its own param
+            reply_to = headers.pop('Reply-To')
+            self.set_reply_to(parse_address_list([reply_to]))
+        if len(headers) > 0:
+            self.data["headers"] = dict(headers)  # flatten to normal dict for json serialization
+        else:
+            del self.data["headers"]  # don't send empty headers
+
+        # SendinBlue use different argument's name if we use template functionality
+        if self.template_id:
+            data = self._transform_data_for_templated_email(self.data)
+        else:
+            data = self.data
+
+        return self.serialize_json(data)
+
+    def _transform_data_for_templated_email(self, data):
+        """
+        Transform the default Payload's data (used for basic transactional email) to
+        the data used by SendinBlue in case of a templated transactional email.
+        :param data: The data we want to transform
+        :return: The transformed data
+        """
+        transformation = {
+            'to': 'emailTo',
+            'cc': 'emailCc',
+            'bcc': 'emailBcc,'
+        }
+        for key in self.data.keys():
+            if key in transformation.keys():
+                new_key = transformation[key]
+                self.data[new_key] = [email.get('email') for email in self.data.pop(key)]
```

Comment:
I think this is going to lose the recipient's display_name when sending a template email.

What you probably want here is the fully-formatted [`email.address`](https://github.com/anymail/django-anymail/blob/v1.4/anymail/backends/mailjet.py#L91-L107) from the original EmailAddress objects passed to `set_recipients`.

Reply:
SendinBlue doesn't support `display_name` on templated email :unamused: ...

https://developers.sendinblue.com/v3.0/reference#sendtemplate
```
curl --request POST \
  --url https://api.sendinblue.com/v3/smtp/templates/templateId/send \
  --data '{"emailTo":["to1@example.com","to2@example.com"]}'
```